controllers/snapshot/snapshot_adapter.go

```diff
@@ -75,6 +76,92 @@ func scenariosNamesToList(integrationTestScenarions *[]v1beta1.IntegrationTestSc
 	return &result
 }
 
+// EnsureRerunPipelineRunsExist is responsible for recreating integration test pipelines triggered by users
+func (a *Adapter) EnsureRerunPipelineRunsExist() (controller.OperationResult, error) {
+
+	scenarioName, ok := gitops.GetIntegrationTestRunLabelValue(*a.snapshot)
+	if !ok {
+		// no test rerun trigerred
+		return controller.ContinueProcessing()
+	}
+	integrationTestScenario, err := a.loader.GetScenario(a.client, a.context, scenarioName, a.application.Namespace)
+
+	if err != nil {
+		if clienterrors.IsNotFound(err) {
+			a.logger.Error(err, "scenario for integration test re-run not found", "scenario", scenarioName)
+			// scenario doesn't exist just remove label and continue
+			if err = gitops.RemoveIntegrationTestRerunLabel(a.client, a.context, a.snapshot); err != nil {
+				return controller.RequeueWithError(err)
+			}
+			return controller.ContinueProcessing()
+		}
+		return controller.RequeueWithError(fmt.Errorf("failed to fetch requested scenario %s: %w", scenarioName, err))
+	}
+
+	a.logger.Info("Re-running integration test for scenario", "scenario", scenarioName)
+
+	testStatuses, err := gitops.NewSnapshotIntegrationTestStatusesFromSnapshot(a.snapshot)
+	if err != nil {
+		return controller.RequeueWithError(err)
+	}
+
+	integrationTestScenarioStatus, ok := testStatuses.GetScenarioStatus(integrationTestScenario.Name)
+	if ok && (integrationTestScenarioStatus.Status == intgteststat.IntegrationTestStatusInProgress ||
```

Comment:
This might be a corner case: Is there a user case that a user wants to rerun an inProgress PLR or trigger PLR for a pending integration test scenario that is not triggered yet due to the error in other integration test scenarios?

Reply:
We have a separate story to run integration tests independently, so one test wouldn't be blocked by error in the other.
I'd not solve this here but rather wait for the other story to be fixed.

Reply:
I am fine with it as long as it is known.